init-odkx-sync-endpoint.py

```diff
@@ -32,11 +39,48 @@ def run_interactive_config():
     typer.echo("We'll need some information from you to get started though...")
     time.sleep(1)
     typer.echo("")
+
+    allow_default = typer.confirm("Do you wish to use default configuratilon?", default=True)
+    typer.echo("")
+
+    if not allow_default:
+        if is_cache_present():
+            allow_cache = typer.confirm("Do you wish to use cached configuratilon?", default=True)
+            typer.echo("")
+            if allow_cache:
+                cached_data = load_progress()
+                enforce_https = json.dumps(cached_data['enforce_https'])
+                manual_certificate = json.dumps(cached_data['manual_certificate'])
+                if manual_certificate : 
+                    setup_certificate(env) 
+                else:
+                    skip_setup_certificate(env)
+
+            else:
+                enforce_https = run_cache_setup(env)
+        else:
+            enforce_https = run_cache_setup(env)
+    else :
+        typer.echo("Please enter your system Password")
+        enforce_https = json.dumps(default_config['enforce_https'])
+        skip_setup_certificate(env)
+
+    write_to_env_file(env_file_location, env)
+
+    return (enforce_https, env)
+
+def run_cache_setup(env : Dict[str, str]) -> bool:
+    allow_save_progress = typer.confirm("Do you wish to save your coniguration?", default=False)
```

Comment:
I think we probably want to always save progress? I think the use default vs. save progress logic could be simplified into just one question to determine if we use the saved progress or reset to default.